src/DI/Extensions/ExtensionsExtension.php

```diff
@@ -30,8 +30,9 @@ public function loadConfiguration()
 				$name = null;
 			}
 			if ($class instanceof Nette\DI\Definitions\Statement) {
-				$rc = new \ReflectionClass($class->getEntity());
-				$this->compiler->addExtension($name, $rc->newInstanceArgs($class->arguments));
+				/** @var Nette\DI\CompilerExtension $rcExtension */
```

Comment:
In fact, we don't know if the `$rcExtension` is CompilerExtension, and it is expected that it might throw a TypeError.

Reply:
Btw, it is better to use `assert($rcExtension instanceof CompilerExtension)`
Locally defined annotations are hack to satisfy IDE

Reply:
Ideally, there should be checking and throwing an exception.

Reply:
`assert` should be used instead of local annotations for internal behavior, to be exact.

Reply:
I implemented exception, but I'm not sure about the message.